Agents/DistrictHeatingOptimisationAgent/README.md

```diff
@@ -0,0 +1,181 @@
+#--- Deployment specific parameters ---#
+# Required for Stack deployment (can be left blank for "standalone" deployment)
+NAMESPACE                     # Blazegraph namespace (within Stack) to monitor
+DATABASE                      # PostGIS/PostgreSQL database name (default: `postgres`)
+# Required for "standalone" deployment (can be left blank for Stack deployment)
+STACK_NAME                    # to be left blank for "standalone" deployment
+DB_URL                        # PostGIS/PostgreSQL URL
+DB_USER                       # PostGIS/PostgreSQL username
+DB_PASSWORD                   # PostGIS/PostgreSQL password
+
+#--- Derivation Agent parameters ---#
+SPARQL_QUERY_ENDPOINT         # SPARQL endpoint to monitor/update
+SPARQL_UPDATE_ENDPOINT        # SPARQL endpoint to monitor/update
+ONTOAGENT_SERVICE_IRI         # IRI of OntoAgent service
+ONTOAGENT_OPERATION_HTTP_URL  # Port needs to match port specified in `docker-compose.yml`
+DERIVATION_INSTANCE_BASE_URL  # Base IRI of all instanced generated by agent
+DERIVATION_PERIODIC_TIMESCALE # Interval in which to check for updated KG information (in s)
+                              # (irrelevant as DH Optimisation Agent uses synchronous derivations only)
+REGISTER_AGENT                # Boolean flag whether to register agent in KG (`true` required to detect derivations)
+# Required inputs for DerivationAgent to start, although not relevant for DH Optimisation Agent (i.e., shall be left blank)
+KG_USERNAME
+KG_PASSWORD
+FILE_SERVER_ENDPOINT
+FILE_SERVER_USERNAME
+FILE_SERVER_PASSWORD
+```
+
+The `STACK_NAME` variable is used to identify the deployment mode of the agent: In case the `STACK_NAME` is left blank, Postgres and Blazegraph endpoint settings will be taken from the `docker-compose file` values. Otherwise they will be retrieved using the StackClients based on the provided `NAMESPACE` and `DATABASE` variables.
+
+Please note: 
+- When deployed via the Stack manager, the `STACK_NAME` environment variable is set automatically and can be omitted in the config file
+- `SPARQL_QUERY_ENDPOINT` and `SPARQL_UPDATE_ENDPOINT` are both required inputs even for stack deployment (due to default derivation agent config); however, as the actual values for initialising the DH Optimisation Agent are retrieved via the Stack Clients, they can be left blank.
+
+## 1.2 Miscellaneous
+
+**Only relevant** if you intend to build (and publish) the Docker image: Ensure access to Github container registry, as
+- the required `stack-clients-*.jar` resource to be added to [py4jps] during building the Docker image is retrieved from the published Stack-Clients docker image
+- a `publish_docker_image.sh` convenience script is provided to build and publish the agent image to the [Github container registry]. To publish a new image, your github user name and [personal access token] (which must have a `scope` that [allows you to publish and install packages]) needs to be provided. 
+
+## 1.3 Required Derivation Markup
```

Comment:
This needs elaboration in my view. As it stands, I have no idea what to do. The heading of 1.3 suggests one needs to instantiate derivations as a prerequisite to make it work, yet the content of the section makes no mention of that. Please expand/add/clarify (in conjunction with more details on how to run it, e.g. how do I tell the agent which instances to run it on).

Reply:
Addressed by [b6d9c3c](https://github.com/cambridge-cares/TheWorldAvatar/pull/1031/commits/b6d9c3cf9df88d31bedcdff24afb653224a127c0)